helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterContext.java

```diff
@@ -55,12 +55,15 @@
   /**
    * Construct the cluster context based on the current instance status.
    * @param replicaSet All the partition replicas that are managed by the rebalancer
-   * @param instanceCount The count of all the active instances that can be used to host partitions.
+   * @param nodeSet All the active nodes that are managed by the rebalancer
    */
-  ClusterContext(Set<AssignableReplica> replicaSet, int instanceCount,
+  ClusterContext(Set<AssignableReplica> replicaSet, Set<AssignableNode> nodeSet,
       Map<String, ResourceAssignment> baselineAssignment, Map<String, ResourceAssignment> bestPossibleAssignment) {
+    int instanceCount = nodeSet.size();
     int totalReplicas = 0;
     int totalTopStateReplicas = 0;
+    Map<String, Integer> totalUsage = new HashMap<>();
+    Map<String, Integer> totalCapacity = new HashMap<>();
 
     for (Map.Entry<String, List<AssignableReplica>> entry : replicaSet.stream()
         .collect(Collectors.groupingBy(AssignableReplica::getResourceName))
```

Comment:
Do we even need to group by resourceName? How does that affect our mappings?

Reply:
Let's sync offline. Too many details.

Reply:
In short, this is for calculating _estimatedMaxPartitionByResource.